src/components/video-controller.js

```diff
@@ -319,27 +419,54 @@ export class VideoController {
         if (adBreak) {
             adBreak.completed = true;
 
-            console.log(`ad break skipped: ${adBreak.id} to: ${this.timeDebugDisplay(adBreak.endTime)}`);
+            console.log(`ad break ${adBreak.index} skipped to: ${this.timeDebugDisplay(adBreak.endTime)}`);
+
+            this.hideControlBar();
 
             // skip a little past the end to avoid a flash of the final ad frame
-            this.seekTo(adBreak.endTime+1, this.isControlBarVisible);
+            this.seekTo(adBreak.endTime + 1, this.isControlBarVisible);
         }
     }
 
-    startAd(adBreak) {
-        if (adBreak.started || adBreak.completed) return false;
+    startAd(googleAd) {
+        const podInfo = googleAd.getAdPodInfo();
+        const adBreak = this.adBreaks[podInfo.getPodIndex()];
+        if (!adBreak) return;
+        if (googleAd.getAdSystem() != 'trueX') return; // ignore non-trueX ads
+
+        // For true[X] IMA integration, the first ad in an ad break points to the interactive ad,
+        // the remaining ones are the fallback ad videos.
+        const adPosition = podInfo.getAdPosition();
+        if (adPosition != 1) return;
+
+        if (adBreak.started) return;
+
+        if (adBreak.completed) {
+            this.skipAd(adBreak);
+            return;
+        }
+
+        var vastConfigUrl = googleAd.getDescription();
+        vastConfigUrl = vastConfigUrl && vastConfigUrl.trim();
+        if (!vastConfigUrl) return;
+        if (!vastConfigUrl.startsWith('http')) {
+            vastConfigUrl = 'https://' + vastConfigUrl;
+        }
+
         adBreak.started = true;
-        console.log(`ad started: ${adBreak.id} at: ${this.timeDebugDisplay(adBreak.startTime)}`);
+        console.log(`truex ad started at ${this.timeDebugDisplay(adBreak.startTime)}:\n${vastConfigUrl}`);
 
         // Start an interactive ad.
         this.hideControlBar();
 
         this.stopVideo(); // avoid multiple videos, e.g. for platforms like the PS4
 
-        // ensure main video is logically at the ad start for when it resumes
-        this.initialVideoTime = adBreak.startTime;
+        // ensure main video is logically at the fallback videos for when it resumes
+        // We just need to skip over the placeholder video of this interactive ad wrapper.
+        adBreak.placeHolderDuration = googleAd.getDuration();
+        this.initialVideoTime = adBreak.fallbackStartTime;
```

Comment:
how about.. linearAdStartTime?

Reply:
well, I think of the remaining videos as the fallback ad videos, to use when the interactive ad cannot be, or is abandoned.